operator/internal/controller/konfluxinfo_controller.go

```diff
@@ -0,0 +1,499 @@
+}
+
+type integrationsJSON struct {
+	GitHub          *gitHubIntegrationJSON `json:"github,omitempty"`
+	SBOMServer      *sbomServerJSON        `json:"sbom_server,omitempty"`
+	ImageController *imageControllerJSON   `json:"image_controller,omitempty"`
+}
+
+type gitHubIntegrationJSON struct {
+	ApplicationURL string `json:"application_url"`
+}
+
+type sbomServerJSON struct {
+	URL     string `json:"url"`
+	SBOMSha string `json:"sbom_sha"`
+}
+
+type imageControllerJSON struct {
+	Enabled       bool               `json:"enabled"`
+	Notifications []notificationJSON `json:"notifications,omitempty"`
+}
+
+type notificationJSON struct {
+	Title  string               `json:"title"`
+	Event  string               `json:"event"`
+	Method string               `json:"method"`
+	Config apiextensionsv1.JSON `json:"config"`
+}
+
+type rbacRoleJSON struct {
+	DisplayName string      `json:"displayName"`
+	Description string      `json:"description"`
+	RoleRef     roleRefJSON `json:"roleRef"`
+}
+
+type roleRefJSON struct {
+	APIGroup string `json:"apiGroup"`
+	Kind     string `json:"kind"`
+	Name     string `json:"name"`
+}
+
+// convertIntegrations converts API types to JSON types
+func convertIntegrations(integrations *konfluxv1alpha1.IntegrationsConfig) *integrationsJSON {
+	if integrations == nil {
+		return nil
+	}
+
+	result := &integrationsJSON{}
+	if integrations.GitHub != nil {
+		result.GitHub = &gitHubIntegrationJSON{
+			ApplicationURL: integrations.GitHub.ApplicationURL,
+		}
+	}
+	if integrations.SBOMServer != nil {
+		result.SBOMServer = &sbomServerJSON{
+			URL:     integrations.SBOMServer.URL,
+			SBOMSha: integrations.SBOMServer.SBOMSha,
+		}
+	}
+	if integrations.ImageController != nil {
+		result.ImageController = &imageControllerJSON{
+			Enabled: integrations.ImageController.Enabled,
+		}
+		if len(integrations.ImageController.Notifications) > 0 {
+			result.ImageController.Notifications = make([]notificationJSON, len(integrations.ImageController.Notifications))
+			for i, n := range integrations.ImageController.Notifications {
+				result.ImageController.Notifications[i] = notificationJSON{
+					Title:  n.Title,
+					Event:  n.Event,
+					Method: n.Method,
+					Config: n.Config, // apiextensionsv1.JSON can be used directly
+				}
+			}
+		}
+	}
+	return result
+}
+
+// convertRBACRoles converts API types to JSON types
+func convertRBACRoles(roles []konfluxv1alpha1.RBACRole) []rbacRoleJSON {
+	result := make([]rbacRoleJSON, len(roles))
+	for i, role := range roles {
+		displayName := role.DisplayName
+		if displayName == "" {
+			displayName = role.Name
+		}
+		result[i] = rbacRoleJSON{
+			DisplayName: displayName,
+			Description: role.Description,
+			RoleRef: roleRefJSON{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     role.Name,
+			},
+		}
+	}
+	return result
+}
+
+// getDefaultRBACRoles returns the default RBAC roles
+func getDefaultRBACRoles() []rbacRoleJSON {
+	return []rbacRoleJSON{
```

Comment:
We already have those defined in the operator, whats the reason for creating them?